### PR TITLE
Migration to rebar3.

### PR DESCRIPTION
**Pull request by walter-weinmann**

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
+
+{deps_error_on_conflict, true}.
+
+{eunit_opts, [
+    {skip_deps, true},
+    verbose
+]}.
+
+{xrl_opts, [dfa_graph, verbose]}.
+
```
Comment (c-bik) [inline rebar.config]: Why `XREF` options aren't required anymore?

File: src/jpparse.app.src [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+            kernel,
+            stdlib
+        ]},
+        {description, "JSON path parser and compiler"},
+        {modules, [
+            jpparse,
+            jpparse_fold
+        ]},
+        {vsn, "0.3.0"}
+    ]}.
```
Comment (c-bik) [inline src/jpparse.app.src]: Can we still maintain `description` and `vsn` as first two properties?

File: src/jpparse.yrl [hunk 3]
```diff
@@ -55,8 +55,7 @@ leaf -> STRING          : unwrap('$1').
 Erlang code.
 
 % parser and compiler interface
--export([parsetree/1, parsetree_with_tokens/1, foldtd/3, foldbu/3, string/1,
-         roots/1]).
+-export([parsetree/1, parsetree_with_tokens/1]).
```
Comment (c-bik) [inline src/jpparse.yrl]: Breaking https://github.com/K2InformaticsGmbH/imem/blob/d35f2832b7f99e7cb2dc9524275108361bde6991/src/imem_json.erl#L296

File: src/jpparse_fold.erl [hunk 4]
```diff
@@ -0,0 +1,107 @@
+-module(jpparse_fold).
+
+-export([foldtd/3, foldbu/3, string/1,
```
Comment (c-bik) [inline src/jpparse_fold.erl]: till 80 characters same line is preferred

File: src/jpparse_fold.erl [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                             ParseTree :: tuple()) -> any().
+foldtd(Fun, Acc, Pt) when is_function(Fun, 3) ->
+    fold_i({td,Fun}, Acc, Pt).
+
+% Folds a jppath parsee tree applying fold function
+%  in a Boottom-Up walk
+-spec foldbu(Function :: fun((Depth :: integer(), SubParseTree :: any(),
+                              AccIn :: any()) -> AccOut :: any()),
+                             Acc :: any(),
+                             ParseTree :: tuple()) -> any().
```
Comment (c-bik) [inline src/jpparse_fold.erl]: `Acc` and `ParseTree` are parameters of `foldbu` vertically aligning them with `Function` parameter is preferred!

File: src/jpparse.yrl (hunk 3 above)
Comment (walter-weinmann) [inline src/jpparse.yrl]: The change is considered in the rebar3 version of imem.

File: rebar.config (hunk 1 above)
Comment (walter-weinmann) [inline rebar.config]: In my understanding xref is only relevant for imem anf dderl.

**Pull request closed without merging**